core/src/test/java/com/datastax/oss/driver/internal/core/context/StartupOptionsBuilderTest.java

```diff
@@ -87,22 +61,43 @@ private void assertDefaultStartupOptions(Startup startup) {
   }
 
   @Test
-  public void should_build_minimal_startup_options() {
-    buildDriverContext();
-    Startup startup = new Startup(defaultDriverContext.getStartupOptions());
+  public void should_build_startup_options_with_no_compression_if_undefined() {
+
+    DefaultDriverContext ctx = MockedDriverContextFactory.defaultDriverContext();
+    Startup startup = new Startup(ctx.getStartupOptions());
     assertThat(startup.options).doesNotContainKey(Startup.COMPRESSION_KEY);
     assertDefaultStartupOptions(startup);
   }
 
   @Test
-  public void should_build_startup_options_with_compression() {
-    when(defaultProfile.isDefined(DefaultDriverOption.PROTOCOL_COMPRESSION))
-        .thenReturn(Boolean.TRUE);
-    when(defaultProfile.getString(DefaultDriverOption.PROTOCOL_COMPRESSION)).thenReturn("lz4");
-    buildDriverContext();
-    Startup startup = new Startup(defaultDriverContext.getStartupOptions());
-    // assert the compression option is present
-    assertThat(startup.options).containsEntry(Startup.COMPRESSION_KEY, "lz4");
+  public void should_build_startup_options_with_no_compression_if_defined_as_none() {
+
+    DefaultDriverContext ctx = buildMockedContext("none");
+    Startup startup = new Startup(ctx.getStartupOptions());
+    assertThat(startup.options).doesNotContainKey(Startup.COMPRESSION_KEY);
     assertDefaultStartupOptions(startup);
   }
+
+  @Test
+  public void should_build_startup_options_with_lz4_compression() {
+
+    doCompressorStartupOptionsTest("lz4");
+  }
+
+  @Test
+  public void should_build_startup_options_with_snappy_compression() {
+
+    doCompressorStartupOptionsTest("snappy");
+  }
+
+  @Test
+  public void should_build_startup_options_with_invalid_compression() {
```

Comment:
Nit: maybe this should be called `should_fail_to_build...`

Reply:
Agreed... I think that's probably a more descriptive name.